src/Microsoft.Azure.WebJobs.Host/Bindings/Path/BindingTemplate.cs

```diff
@@ -54,22 +53,36 @@ internal IEnumerable<BindingTemplateToken> Tokens
         /// <summary>
         /// Gets the collection of parameter names this pattern applies to.
         /// </summary>
-        public IEnumerable<string> ParameterNames
-        {
-            get
-            {
-                return from p in Tokens
-                       let name = p.ParameterName
-                       where name != null
-                       select name;
-            }
-        }
+        public IEnumerable<string> ParameterNames => Tokens.Where(p => p.ParameterName != null).Select(p => p.ParameterName);
+
+        /// <summary>
+        /// Gets the collection of token strings in the pattern.
+        /// </summary>
+        public IEnumerable<string> TokenStrings => Tokens.Where(p => p.TokenString != null).Select(p => p.TokenString);
 
         /// <summary>
         /// True if this expression has parameters. 
         /// </summary>
         public bool HasParameters => ParameterNames.Any();
 
+        /// <summary>
+        /// A utility for resolving individual tokens against binding data.
+        /// </summary>
+        /// <param name="token">A token string to resolve from the binding data. For example, "a.b.c".</param>
+        /// <param name="bindingData">The binding data to resolve the parameter against.</param>
+        /// <returns>The resolved parameter.</returns>
+        public static string ResolveToken(string token, IReadOnlyDictionary<string, object> bindingData)
```

Comment:
Is this equivalent to the below? Wondering who is going to use this new method and if the below isn't sufficient.

```csharp
var template = BindingTemplate.FromString(token);
template.Bind(bindingData);
```